python-avd/pyavd/_eos_designs/shared_utils/wan.py

```diff
@@ -542,6 +542,41 @@ def wan_ha_interfaces(self: SharedUtils) -> list:
         # Using node values
         return natural_sort(set(self.configured_wan_ha_interfaces), "name")
 
+    @cached_property
+    def wan_ha_port_channel_id(self: SharedUtils) -> int:
+        """Port-channel ID to use for direct WAN HA port-channel.
+
+        If not provided, computed from the list of configured members.
+        """
+        if self.use_uplinks_for_wan_ha:
+            msg = "Using Port-Channel when using uplinks as HA is not supported."
+            raise AristaAvdError(msg)
+        default_wan_ha_port_channel_id = int("".join(findall(r"\d", self.wan_ha_interfaces[0])))
+        return get(self.switch_data_combined, "wan_ha.port_channel_id", default_wan_ha_port_channel_id)
+
+    @cached_property
+    def use_port_channel_for_direct_ha(self: SharedUtils) -> bool:
+        """Indicate if port-channel should be used for direct HA.
+
+        Returns:
+            bool: False is use_uplinks_for_wan_ha is True, otherwise the value of `wan_ha.use_port_channel_for_direct_ha`
+                  which defaults to True.
+
+        Raises:
+            AristaAvdError: When `wan_ha.use_port_channel_for_direct_ha` is False and there are multiple configured interfaces.
+        """
+        if self.use_uplinks_for_wan_ha:
+            return False
+
+        use_port_channel = get(self.switch_data_combined, "wan_ha.use_port_channel_for_direct_ha", True)
+        interfaces = set(self.configured_wan_ha_interfaces)
+
+        if len(interfaces) > 1 and use_port_channel is False:
```

Comment:
your schema description on the use_port_channel knob suggests that this only applies when there is a single interface. So we should either change the description, or just ignore this knob when we have multiple interfaces. Better than raising, negative tests etc.

Reply:
will ignore the knob

Reply:
removed negative unit test as well